soh/src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -427,7 +427,8 @@ u16 func_80B61024(PlayState* play, Actor* thisx) {
                 return 0x402D;
             }
 
-            if (Flags_GetEventChkInf(EVENTCHKINF_KING_ZORA_MOVED)) {
+            if (CVarGetInteger("gFixZoraHintDialogue", 0) && !Flags_GetEventChkInf(EVENTCHKINF_KING_ZORA_MOVED) ||
+                !CVarGetInteger("gFixZoraHintDialogue", 0) && Flags_GetEventChkInf(EVENTCHKINF_KING_ZORA_MOVED)) {
```

Comment:
I think the fix should incorporate checking that the player has rutos letter `EVENTCHKINF_OBTAINED_RUTOS_LETTER`, otherwise the condition below will trigger out of order. `0x400F` is what tells the player a hint to "check lake hylia".

I'm not sure if we should still check for moving king zora as part of the fix. Maybe having letter is enough. That way the zora uses this text until you get the zoras sapphire.